src/mcp23017.cpp

```diff
@@ -1,33 +1,96 @@
 #include "mcp23017.h"
-#include <errno.h>
-
 
 // TODO (optional): Define macros for useful register below:
-
+#define IODIRA 0x00
+#define IODIRB 0x01
+#define GPIOA 0x12
+#define GPIOB 0x13
 
 // TODO: Initialize i2cBus member
-Mcp23017::Mcp23017(int addr, I2C* i2cBus) {
-
+Mcp23017::Mcp23017(int addr) {
+    this->addr = addr;
 }
 
 uint8_t Mcp23017::get_dir(int pin) {
```

Comment:
Whenever you read from a register of the IO Expander, you need to write the register address to the IO expander before you can read the value in the register from the IO Expander. See this [Confluence page on I2C reads](https://badgerloop.atlassian.net/wiki/spaces/EL/pages/279281763/Serial+Protocols+with+MBedOS#I2C-read) for more details. This applies for all functions where you do an I2C read.